### PR TITLE
Report deprecations for deprecated members from the org.gradle.util package

### PR DESCRIPTION
**Pull request by eskatos**
* https://github.com/gradle/gradle/issues/21839

For some members of `org.gradle.util.*` deprecation logging could not be added because smoke tested plugins and other third parties don't offer a version that is free of their use:

* **Kotlin Gradle Plugin**
  * `WrapUtil.toDomainObjectSet()`
  * [KT-53882 - Kotlin Gradle Plugin should not use the deprecated WrapUtil.toDomainObjectSet() function](https://youtrack.jetbrains.com/issue/KT-53882/Kotlin-Gradle-Plugin-should-not-use-the-deprecated-WrapUtiltoDomainObjectSet-function)
  * https://github.com/gradle/gradle/issues/21882
* **Android Gradle Plugin**
  * `GUtil.toWords(string)`
  * [ANDROID-245405989 - Android Gradle Plugin should not use the deprecated GUtil.toWords(string) function](https://issuetracker.google.com/issues/245405989)
  * https://github.com/gradle/gradle/issues/21899
  * `ConfigureUtil.configure(closure, target)`
  * [ANDROID-245405989 - Android Gradle Plugin should not use the deprecated ConfigureUtil.configure(closure, target) function](https://issuetracker.google.com/issues/245405994)
  * https://github.com/gradle/gradle/issues/21901
* **Spring Dependency Management Plugin**
  * `GradleVersion.getNextMajor()`
  * https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/338
  * https://github.com/gradle/gradle/issues/21898
* **Google Protobuf Gradle Plugin**
  * `GUtil.toCamelCase(string)`
  * https://github.com/google/protobuf-gradle-plugin/pull/604
  * https://github.com/gradle/gradle/issues/21900
  * `ConfigureUtil.configure(closure, target)`
  * https://github.com/google/protobuf-gradle-plugin/pull/577
  * https://github.com/gradle/gradle/issues/21901
* **OSDetector Gradle Plugin (via the Protobuf plugin)**
  * `VersionNumber.*`
  * https://github.com/google/osdetector-gradle-plugin/issues/24
  * https://github.com/gradle/gradle/issues/21902
* **Asciidoctor Gradle Plugin / Grolifant**
  * `CollectionUtils.stringize(collection)`
  * [GROLIFANT-85 - Grolifant should not use the deprecated CollectionUtils](https://gitlab.com/ysb33rOrg/grolifant/-/issues/85)
  * https://github.com/gradle/gradle/issues/21903
* **Nebula Lint Plugin**
  * `GUtil.loadProperties(inputStream)`
  * https://github.com/nebula-plugins/gradle-lint-plugin/issues/378
  * https://github.com/gradle/gradle/issues/21912
* **Nebula Dependency Lock Plugin**
  * `NameMatcher`
  * https://github.com/nebula-plugins/gradle-dependency-lock-plugin/issues/243
  * https://github.com/gradle/gradle/issues/21913
* **IntelliJ IDEA**
  * `GUtil.toCamelCase(string)`
  * `GUtil.toLowerCamelCase(string)`
  * `GUtil.loadProperties(inputStream)`
  * `ConfigureUtil.configureByMap(properties, delegate)`
  * [IDEA-301430 - IntelliJ IDEA should not use deprecated types from the org.gradle.util package](https://youtrack.jetbrains.com/issue/IDEA-301430/IntelliJ-IDEA-should-not-use-deprecated-types-from-the-orggradleutil-package)
  * https://github.com/gradle/gradle/issues/21912
  * https://github.com/gradle/gradle/issues/21900
  * https://github.com/gradle/gradle/issues/21914

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test AST

**Comment by eskatos** [comment]
@bot-gradle test AST

**Comment by eskatos** [comment]
@bot-gradle test AST

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test AFT

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Review by nathan-contino — APPROVED** [review]
Docs LGTM

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/logging/src/main/java/org/gradle/util/CollectionUtils.java [hunk 1]
```diff
@@ -58,23 +59,41 @@
...
     /**
      * Returns the single element in the collection or throws.
      */
     public static <T> T single(Iterable<? extends T> source) {
+        logDeprecation();
+        return singleInternal(source);
+    }
+
+    private static <T> T singleInternal(Iterable<? extends T> source) {
+        logDeprecation();
```
Comment (ljacomet) [inline subprojects/logging/src/main/java/org/gradle/util/CollectionUtils.java:96]: All calls are protected with logging. So this looks like a duplicate.
Shouldn't we either log here and not in the callers or not log here?

The later would match what you did for `collectInternal`.

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall.
Have two change requests and a couple comments / questions

File: subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/ToolingApiClasspathIntegrationTest.groovy [hunk 2]
```diff
@@ -34,6 +34,6 @@ class ToolingApiClasspathIntegrationTest extends AbstractIntegrationSpec {
         resolve.classpath.any {it.name ==~ /slf4j-api-.*\.jar/}
         // If this suddenly fails without an obvious reason, you likely have added some code
         // that references types that were previously eliminated from gradle-tooling-api.jar.
-        resolve.classpath.find { it.name ==~ /gradle-tooling-api.*\.jar/ }.size() < 2_500_000
+        resolve.classpath.find { it.name ==~ /gradle-tooling-api.*\.jar/ }.size() < 2_600_000
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/ToolingApiClasspathIntegrationTest.groovy:37]: Yes, this is the `org.gradle.util` code moved to `:logging` that made the tapi jar reach the size threshold

File: subprojects/logging/src/main/java/org/gradle/util/internal/DefaultGradleVersion.java [hunk 3]
```diff
@@ -186,6 +187,10 @@ public String getVersion() {
     @Override
     @Deprecated
     public String getBuildTime() {
+        DeprecationLogger.deprecateMethod(GradleVersion.class, "getBuildTime()")
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/logging/src/main/java/org/gradle/util/internal/DefaultGradleVersion.java]: Good point, did just that

File: subprojects/logging/src/main/java/org/gradle/util/CollectionUtils.java (hunk 1 above)
Comment (eskatos) [reply subprojects/logging/src/main/java/org/gradle/util/CollectionUtils.java]: Nice catch. Did the later

**Comment by eskatos** [comment]
Thanks Louis. PTAL

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test AST

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Review by ljacomet — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**